src/functionalTest/kotlin/com/github/jengelman/gradle/plugins/shadow/caching/TransformerCachingTest.kt

```diff
@@ -145,6 +145,40 @@ class TransformerCachingTest : BaseCachingTest() {
     assertions("baz")
   }
 
+  @Test
+  fun shadowJarIsCachedCorrectlyWhenUsingPropertiesFileTransformer() {
+    projectScriptPath.appendText(
+      transform<PropertiesFileTransformer>(
+        transformerBlock = """
+          keyTransformer = { it.toUpperCase() }
+        """.trimIndent(),
+      ),
+    )
+    path("src/main/resources/foo/bar.properties").writeText("foo=bar")
+    val assertions = { key: String ->
+      assertThat(outputShadowJar).useAll {
+        containsEntries("shadow/Main.class", "foo/bar.properties")
+        getContent("foo/bar.properties").contains("$key=bar")
+      }
+    }
+
+    assertFirstExecutionSuccess()
+    assertions("FOO")
```

Comment:
```
expected to contain:<"FOO=bar"> but was:<"#

foo=bar
"> (com.github.jengelman.gradle.plugins.shadow.util.JarPath@5827af16)
org.opentest4j.AssertionFailedError: expected to contain:<"FOO=bar"> but was:<"#

foo=bar
```

This failure looks due to `keyTransformer` being `@Transient`, can't be serialized back, it's null in 

https://github.com/GradleUp/shadow/blob/58564d37edff1ba1b913d3ba6bc4557d828b96b0/src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/PropertiesFileTransformer.kt#L179-L181

Reply:
I tried a serializable type in d71e8790b8b69c22d781e3e254518b89cb29d595, but with a new error

```

* What went wrong: |  
-- | --
  | com/github/jengelman/gradle/plugins/shadow/transformers/KeyTransformer |  
  | > com.github.jengelman.gradle.plugins.shadow.transformers.KeyTransformer |  
  |   |  
  | * Try: |  
  | > Run with --info or --debug option to get more log output. |  
  | > Run with --scan to get full insights. |  
  | > Get more help at https://help.gradle.org. |  
  |   |  
  | * Exception is: |  
  | java.lang.NoClassDefFoundError: com/github/jengelman/gradle/plugins/shadow/transformers/KeyTransformer
```

No idea how to fix them now...